test/jeison-test.el

```diff
@@ -326,20 +317,43 @@
                 (and (not (null client-1))
                      (not (null (gethash 'codemodel-v2 client-1)))
                      (not (null client-2))
-                     (null (gethash 'non-existent client-table))
-
-                     ;; All of these tests succeed: puthash isn't picky about
-                     ;; what it stashes. The caveat: the hash table uses 'eq'
-                     ;; for comparisons.
-                     (puthash "client-123456" "client-123456 data" client-table)
-                     (puthash "client-7890ab" [1 2 3 4 5] client-table)
-                     (puthash [1 3 5] "vector data" client-table)
-                     ;; These 'gethash'-es fail because strings don't compare
-                     ;; with 'eq'.
-                     (null (gethash "client-123456" client-table))
-                     (null (gethash "client-7890ab" client-table))
-                     ;; And neither do vectors.
-                     (null (gethash [1 3 5] client-table))))))))
+                     (null (gethash 'non-existent client-table))))))))
+
+(ert-deftest jeison:check-invalid-hash-key-spec ()
+  ;; Ensure jeison-read's hash table key type checking doesn't admit something
+  ;; other than string or symbol.
+  (jeison-defclass jeison:integer-hash-key-class ()
+    ((a :initarg :a :type (jeison-hash-table-of integer string))))
+  (jeison-defclass jeison:vector-hash-key-class ()
+    ((a :initarg :a :type (jeison-hash-table-of vector string))))
+  (should (and
+           (condition-case nil
+              (progn
+                (jeison-read jeison:integer-hash-key-class "{ \"key\": \"value\" }")
+                (ert-fail "jeison-read admitted hash table with integer key."))
+             (jeison-invalid-key-type t))
+           (condition-case nil
+              (progn
+                (jeison-read jeison:vector-hash-key-class "{ \"key\": \"value\" }")
+                (ert-fail "jeison-read admitted hash table with vector key."))
+             (jeison-invalid-key-type t)))))
+
+(ert-deftest jeison:wrong-hash-value-type ()
+  ;; Exercise 'jeison-wrong-parsed-type' with hash tables.
+  (jeison-defclass jeison:hash-value-entity ()
+    ((a :initarg :a :initform "default" :type string)
+     (b :initarg :b :initform 'undefined :type symbol)
+     (c :initarg :c :initform [] :type (jeison-vector-of integer))))
+  (let ((json-stream "{ \"key-1\": { \"a\" : \"a's value\", \"b\": \"b-symbol-value\", \"c\": [99, 100, 101] } }"))
```

Comment:
Good, this test matches one of the tests that I asked for. However, the input here still looks like a hash-table, so it checks if the values of something that looks like a hash-table match user expectations. I asked for another test, where the value in the specified path DOES NOT LOOK like hash-table, ie is not parsed as an `alist`.